### PR TITLE
fix: #584 change fit-tool to garmin-fit-sdk

### PR DESCRIPTION
**Pull request by ben-29**
using Garmin official sdk to read fit file, fixing #584

### CODE REVIEW

**Review by NaturezzZ — COMMENTED** [review]
Should we add activities without .fit into the running page? For example, some activities manually uploaded with gpx.

**Comment by ben-29** [comment]
> Should we add activities without .fit into the running page? For example, some activities manually uploaded with gpx.

I understand. When downloading the .fit file from Garmin, it returns a zip file containing the GPX if there is no .fit file available. Since it already downloads the GPX, I believe it's better to import it. I'll modify the code to handle this.

**Comment by ben-29** [comment]
@NaturezzZ Could you help me verify if the issue has been resolved?

**Review by NaturezzZ — APPROVED** [review]
I fixed a small import bug. Now LGTM.

File: run_page/gpxtrackposter/track.py [hunk 1]
```diff
@@ -31,6 +25,7 @@
 run_map = namedtuple("polyline", "summary_polyline")
 
 IGNORE_BEFORE_SAVING = os.getenv("IGNORE_BEFORE_SAVING", False)
+SEMICIRCLE = 11930465
```
Comment (yihong0618) [inline run_page/gpxtrackposter/track.py:28]: add comment here for this number

File: run_page/gpxtrackposter/track.py [hunk 2]
```diff
@@ -91,9 +86,12 @@ def load_fit(self, file_name):
             # (for example, treadmill runs pulled via garmin-connect-export)
             if os.path.getsize(file_name) == 0:
                 raise TrackLoadError("Empty FIT file")
-
-            fit = FitFile.from_file(file_name)
-            self._load_fit_data(fit)
+            stream = Stream.from_file(file_name)
+            decoder = Decoder(stream)
+            messages, errors = decoder.read(convert_datetimes_to_dates=False)
+            if len(errors) > 0:
```
Comment (yihong0618) [inline run_page/gpxtrackposter/track.py:92]: if errors:

**Comment by ben-29** [comment]
@NaturezzZ I brought the fit-tool back because the Garmin SDK doesn't offer the "encode to fit" function. Otherwise, "wrap_device_info" wouldn't work for Strava. Could you please take a look? Thanks

@yihong0618 Changed.

**Comment by NaturezzZ** [comment]
okay👌 it looks good to me

**Review by yihong0618 — APPROVED** [review]

**Pull request merged**